### PR TITLE
Astro 3444 clock datein

### PR DESCRIPTION
**Pull request by micahjones13**
## Brief Description

Adds a `date-in` prop to rux-clock that allows the user to set the date to be used by clock. 

Adds a snapshot test, storybook story variant as well.

## JIRA Link

https://rocketcom.atlassian.net/browse/ASTRO-3444

## Related Issue
https://github.com/RocketCommunicationsInc/astro/issues/384
## General Notes

Adds additional logic into the _updateTime function if `date-in` is provided. I left my comments in for now to better show my thinking behind it, I will remove them once this get's approved. 

The story is just called `With Date In` but we could call it `With Custom Date` or something. I'm fine either way

## Motivation and Context

Feat request - allow user to choose a date for clock to go by, rather than a date.now() 

## Issues and Limitations

Similar to AOS/LOS, the timestamp works best with the offset provided - if not it still uses the local timezone.

## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Checklist

- [x] This PR adds or removes a Storybook story.
- [x] I have added tests to cover my changes.
- [ ] Regressions are passing and/or failures are documented
- [x] Changes have been checked in evergreen browsers

### CODE REVIEW

File: packages/web-components/src/components/rux-button/readme.md [hunk 1]
```diff
@@ -143,27 +143,6 @@ For more information about AstroUXDS usage outside of a Web Component environmen
 | `"icon"`      | the optional rux-icon                 |
 
 
-## CSS Custom Properties
```
Comment (micahjones13) [inline packages/web-components/src/components/rux-button/readme.md:146]: I don't know why these got removed? They're still declared in the SCSS file.

**Review by markacianfrani — CHANGES_REQUESTED** [review]
Can we get this to support a number (milliseconds since whatever, ie Date.now() - 1652122612197)  like the AOS/LOS attr do? 

Might also want to validate an actual date is passed in. with aos/los, its not so bad because it just runs once, but this will throw an error every second.

lookin good tho!

File: packages/web-components/src/components/rux-clock/rux-clock.tsx [hunk 2]
```diff
@@ -119,17 +131,66 @@ export class RuxClock {
...
+            this._validateDateInUnix(this.dateIn!)
+        }
+        this._time = this.dateIn!
+        if (!this._rawTime) this._rawTime = new Date(this.dateIn!)
+        if (this._validateDateIn(this._rawTime)) {
+            this._timer = window.setInterval(() => {
+                this._updateTime()
+            }, 1000)
+        } else {
+            console.warn(
```
Comment (micahjones13) [inline packages/web-components/src/components/rux-clock/rux-clock.tsx:145]: We don't have to do the console warn here, it already will produce a console error. I figured this just tells you exactly what threw the error, but I'm okay with getting rid of it

File: packages/web-components/src/components/rux-clock/rux-clock.tsx [hunk 3]
```diff
@@ -119,17 +131,66 @@ export class RuxClock {
         )
     }
 
+    private _handleDateIn() {
+        if (!this.dateIn!.includes('-')) {
```
Comment (micahjones13) [inline packages/web-components/src/components/rux-clock/rux-clock.tsx:135]: I'm not sure if this is the best way to determine if it's a unix date or not. Any better suggestions?

**Pull request merged**